From cf86593cfb4ef581d6295d9447db585917303e91 Mon Sep 17 00:00:00 2001
From: Razvan A Lupusoru <razvan.a.lupusoru@intel.com>
Date: Thu, 19 Dec 2013 16:24:49 -0800
Subject: Dalvik: Fix PCG implementation of Math.abs(F)F

BZ: 160211

The execute-inline version of Math.abs(F)F for PCG calculates the
absolute value by trying to clear the sign bits. However, the value
used for the "and" operation was zero instead of the correct mask.

Change-Id: I754d300b21c91d4d58264e1ced20c29514bcd625
Category: device enablement
Domain: AOSP.Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, need rework
Signed-off-by: Razvan A Lupusoru <razvan.a.lupusoru@intel.com>
---
 vm/compiler/codegen/x86/pcg/LowerCall.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/codegen/x86/pcg/LowerCall.cpp b/vm/compiler/codegen/x86/pcg/LowerCall.cpp
index 004c512..95354cb 100644
--- a/vm/compiler/codegen/x86/pcg/LowerCall.cpp
+++ b/vm/compiler/codegen/x86/pcg/LowerCall.cpp
@@ -646,7 +646,7 @@ void dvmCompilerPcgTranslateExecuteInline (CompilationUnitPCG *cUnit, MIR *mir)
         case INLINE_MATH_ABS_FLOAT:
             {
                 __attribute__ ( (aligned (16))) static u4 mask[4] =
-                    {0, 0, 0, 0x7fffffff};
+                    {0x7fffffff, 0, 0, 0};
 
                 CGInst load = dvmCompilerPcgGetVirtualReg (cUnit,
                     ssaRep->uses[0], "movss1", 4);
-- 
1.7.4.1

